File: python/bubble_sort.py
# Bubble sort

def bubble_sort(arr: list):
    n = len(arr)
    while True:
        swapped = False
        for i in range(n - 1):
            if arr[i] > arr[i + 1]:
                arr[i], arr[i + 1] = arr[i + 1], arr[i]
                swapped = True
        
        if not swapped:
            break
        n -= 1
        
my_list = [4, 1, 0, 3, 2]
my_list = [4, 0, 1, 2, 3]
bubble_sort(my_list)

print(my_list)

File: golang/bubble_sort.go
package main

import "fmt"

func bubbleSort(arr []int) {
	n := len(arr)
	for {
		swapped := false
		for i := 0; i < n-1; i++ {
			if arr[i] > arr[i+1] {
				arr[i], arr[i+1] = arr[i+1], arr[i]
				swapped = true
			}
		}
		if !swapped {
			break
		}
		n -= 1
	}
}

func callBubbleSort() {
	list := []int{4, 0, 1, 2, 3}
	bubbleSort(list)

	fmt.Println("Bubble sort:", list)
}
